### PR TITLE
Scottx611x/acceptance testing

### PR DESCRIPTION
**Pull request by scottx611x**
Introduce `StaticLiveServerTestCases` and `factory_boy` object creation as a solid foundation for writing more acceptance tests

Refactor older tests that interact with the ORM to inherit from `TransactionTestCase` to ensure a clean db after each test run, and to move away from the use of [large tearDown() methods.](https://github.com/refinery-platform/refinery-platform/blob/develop/refinery/core/tests.py#L1315)

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,16 +1,21 @@
...
   - postgresql
   - neo4j
 install:
   - pip install -r requirements.txt --quiet
 env:
-  - DJANGO_SETTINGS_MODULE=config.settings.dev REDUCE_TEST_OUTPUT=true PYTHONPATH=$PYTHONPATH:../refinery:../refinery/config
+  global:
+    # These env vars are available to every build
+    - REDUCE_TEST_OUTPUT=true
+    - PYTHONPATH=$PYTHONPATH:../refinery:../refinery/config
```
Comment (scottx611x) [inline .travis.yml]: Remove redundant PYTHONPATH

File: selenium/acceptance_tests.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+# Fail script if error occurs at any step
+set -e
+
+echo "Setting up env vars..."
+export PYTHONPATH=$PYTHONPATH:../refinery:../refinery/config
+export BASE_URL=http://localhost:8000 CREDS_YML=guest_creds.yml UPLOAD=gff3.csv
+
+echo "Installing Geckodriver..."
+TAR=geckodriver-v0.11.1-linux64.tar
+wget https://github.com/mozilla/geckodriver/releases/download/v0.11.1/$TAR.gz > acceptance_tests.log 2>&1
```
Comment (scottx611x) [inline selenium/acceptance_tests.sh]: `curl -o` ??

File: refinery/factory_boy/django_model_factories.py [hunk 3]
```diff
@@ -59,107 +62,31 @@ class Meta:
...
     class Meta:
         model = "core.WorkflowEngine"
 
 
 class WorkflowFactory(factory.DjangoModelFactory):
     """Minimal representation of a Workflow for testing purposes"""
+
     class Meta:
         model = "core.Workflow"
+        django_get_or_create = ('uuid',)
```
Comment (scottx611x) [inline refinery/factory_boy/django_model_factories.py]: `django_get_or_create` - add comments explaining helpfulness

File: refinery/factory_boy/utils.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+def factory_boy_cleanup():
+    """Remove objects created by factory_boy"""
+
+    # DataSet deletion will take care of Analysis deletion as well
+
+    # NOTE: Be wary of adding deletions here as a:
+    # `Thing.objects.all().delete()` could potentially delete a User's data
+    # when running acceptance tests locally.
+
```
Comment (scottx611x) [inline refinery/factory_boy/utils.py]: Really should be running on a separate database!

File: selenium/test_global_analysis_popover.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+from django.contrib.auth.models import User
+
+from core.models import DataSet, Analysis
+from factory_boy.utils import make_analyses_with_single_dataset
+from utils.selenium_utils import (assert_body_text,
+                                  wait_until_id_clickable, DEFAULT_WAIT, login,
+                                  cleanup_on_error)
+
+base_url = os.environ['BASE_URL']
+not_travis = not('TRAVIS' in os.environ and os.environ['TRAVIS'] == 'true')
```
Comment (scottx611x) [inline selenium/test_global_analysis_popover.py]: probably dont need this

File: selenium/test_no_login.py [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import os
+import pytest
+from django.contrib.auth.models import User, Group
+
+from core.models import DataSet, Workflow, Project
+from file_store.models import FileStoreItem
+from utils.selenium_utils import assert_body_text
+
+base_url = os.environ['BASE_URL']
+not_travis = not('TRAVIS' in os.environ and os.environ['TRAVIS'] == 'true')
```
Comment (scottx611x) [inline selenium/test_no_login.py]: remove !

File: selenium/test_ui_deletion.py [hunk 7]
```diff
@@ -34,225 +30,200 @@ def selenium(selenium):
...
+            total_datasets))
 
     # Delete individual Datasets until there are none left
     while total_datasets:
         selenium.find_elements_by_class_name('dataset-delete')[0].click()
 
-        selenium.implicitly_wait(3)
-
-        wait_until_id_clickable(selenium, 'dataset-delete-button', 5).click()
+        wait_until_id_visible(
```
Comment (scottx611x) [inline selenium/test_ui_deletion.py]: vis or click but not both here

File: refinery/config/settings/base.py [hunk 8]
```diff
@@ -577,3 +578,5 @@ def get_setting(name, settings=local_settings):
 REFINERY_TUTORIAL_STEPS = refinery_tutorial_settings
 
 ANONYMOUS_USER_ID = -1
+
+TEST_RUNNER = "django.test.runner.DiscoverRunner"
```
Comment (hackdna) [inline refinery/config/settings/base.py]: Existing configuration has `TEST_RUNNER = 'django_nose.NoseTestSuiteRunner`. Does this mean we don't need Nose and django-nose anymore?

File: deployment/apache.conf.erb [hunk 9]
```diff
@@ -0,0 +1,47 @@
+# Generated from apache.erb.conf
```
Comment (hackdna) [inline deployment/apache.conf.erb]: This file should no longer exist.

File: refinery/config/settings/base.py (hunk 8 above)
Comment (scottx611x) [inline refinery/config/settings/base.py]: @hackdna Not sure, will have to do some reading. This was the docs suggestion to suppress the warnings that system check was throwing on test runs

File: deployment/apache.conf.erb (hunk 9 above)
Comment (scottx611x) [inline deployment/apache.conf.erb]: @hackdna It wont once this branch is up to date with develop

File: refinery/config/settings/base.py (hunk 8 above)
Comment (scottx611x) [inline refinery/config/settings/base.py]: @hackdna: After reading up, `DiscoverRunner` is providing the same functionality that we are currently using Django nose for.

Some of django-nose's other notable features that we would lose are:
```
- Fixture bundling, an optional feature which speeds up your fixture-based tests by a factor of 4
- Reuse of previously created test DBs, cutting 10 seconds off startup time
- Hygienic TransactionTestCases, which can save you a DB flush per test
```

I don't have a strong opinion for the keeping or removal of django-nose.

File: refinery/factory_boy/utils.py [hunk 10]
```diff
@@ -45,52 +28,25 @@ def make_analyses_with_single_dataset(number_to_create, user_instance):
...
+    while number_to_create:
         analysis_uuid = uuid_builtin.uuid4()
         AnalysisFactory(
             uuid=analysis_uuid,
             name="Test Analysis - {}".format(analysis_uuid),
             project=project,
             data_set=dataset,
             workflow=workflow,
-            status=status_choices[analysis_status_counter][0]
+            status=random.choice(status_choices)[0]
```
Comment (scottx611x) [inline refinery/factory_boy/utils.py]: Double check if status field is needed (random is bad)

File: refinery/file_store/migrations/0002_filetypes_and_fileextensions.py [hunk 11]
```diff
@@ -1,633 +1,306 @@
...
+            name="bai",
+            filetype=Filetype.objects.get(name="BAI file")
+        )
     ]
-    data_as_json = json.dumps([ob for ob in data])
-    objects = serializers.deserialize('json', data_as_json, ignorenonexistent=True)
-    for obj in objects:
-        obj.save()
+
+    for file_extension in file_extensions:
```
Comment (scottx611x) [inline refinery/file_store/migrations/0002_filetypes_and_fileextensions.py]: same as above for Ontologies

File: refinery/selenium_testing/tests.py [hunk 12]
```diff
@@ -22,8 +22,8 @@ class SeleniumTestBase(StaticLiveServerTestCase):
     def setUp(self, site_login=True, initialize_guest=True):
         self.browser = webdriver.Firefox()
         self.browser.maximize_window()
-        # Manually create and save public group to sync Selenium and test
-        # threads
+        # Manually create and save public group to sync Selenium and gecko
+        # driver threads
         create_public_group()
         ExtendedGroup.objects.public_group().save()
```
Comment (scottx611x) [inline refinery/selenium_testing/tests.py]: ONe of these lines is not necessary (probably create_public_group() )

**Comment by scottx611x** [comment]
[review and cleanup database schema](https://trello.com/c/Vf8ZRbNF/93-review-and-cleanup-database-schema)

**Comment by scottx611x** [comment]
[review and cleanup database schema](https://trello.com/c/Vf8ZRbNF/93-review-and-cleanup-database-schema)

**Comment by scottx611x** [comment]
[review and cleanup database schema](https://trello.com/c/Vf8ZRbNF/93-review-and-cleanup-database-schema)

**Pull request merged**